client/src/components/InfoCard.tsx

```diff
@@ -1,77 +1,89 @@
-import { useState } from "react";
-import {
-  Avatar,
-  Badge,
-  Box,
-  Button,
-  Card,
-  CardBody,
-  CardFooter,
-  CardHeader,
-  ChakraProvider,
-  Flex,
-  Heading,
-  IconButton,
-} from "@chakra-ui/react";
-import { Autocomplete, Option } from "chakra-ui-simple-autocomplete";
-import { BiLike, BiChat, BiShare } from "react-icons/bi";
+import { Dispatch, SetStateAction, useRef } from "react";
+import { Badge, Box, Button, Image, Textarea } from "@chakra-ui/react";
+import { BiLike, BiChat } from "react-icons/bi";
 import { StarIcon } from "@chakra-ui/icons";
-const InfoCard = () => {
-  const property = {
-    imageUrl: "https://bit.ly/2Z4KKcF",
-    imageAlt: "Rear view of modern home with pool",
-    beds: 3,
-    baths: 2,
-    title: "2445 Guilford dr, Abbotsford BC ",
-    formattedPrice: "$2,900",
-    reviewCount: 20,
-    rating: 4,
+import { Post } from "../types/Post";
+import { Comment } from "../types/Comment";
+import "styles/userHome.css";
+
+interface props {
+  post: Post;
+  comment: Comment;
+  setComment: Dispatch<SetStateAction<Comment>>;
+  updateComments: (e: any) => void;
+}
+const InfoCard = (props: props) => {
+  const inputRef = useRef<any>(null);
+  const toggleCommentForm = () => {
+    if (inputRef.current!.style.display == "block") {
+      inputRef.current!.style.display = "none";
+      return;
+    }
+    if (inputRef.current!.style.display == "") {
+      inputRef.current!.style.display = "block";
+      return;
+    }
+    inputRef.current!.style.display = "block";
   };
```

Comment:
```suggestion
const InfoCard = (props: props) => {
  const inputRef = useRef<HTMLInputElement>(null);
  const toggleCommentForm = () => {
    if (inputRef.current == null) {
      return;
    }

    if (inputRef.current.style.display == "block") {
      inputRef.current.style.display = "none";
      return;
    }
    if (inputRef.current.style.display == "") {
      inputRef.current.style.display = "block";
      return;
    }
    inputRef.current.style.display = "block";
  };
```
Try not to use `any` and non-null assertion. Just do a null check and Typescript will catch on